docs/connecting.asciidoc

```diff
@@ -0,0 +1,145 @@
+[[connecting]]
+== Connecting
+
+This page contains the information you need to create an instance of the .NET 
+Client for {es} that connects to your {es} cluster.
+
+It’s possible to connect to your {es} cluster via a single node, or by 
+specifying multiple nodes using a node pool. Using a node pool has a few 
+advantages over a single node, such as load balancing and cluster failover 
+support. The client provides convenient configuration options to connect to an 
+Elastic Cloud deployment.
+
+IMPORTANT: Client applications should create a single instance of 
+`ElasticsearchClient` that is used throughout your application for its entire 
+lifetime. Internally the client manages and maintains HTTP connections to nodes, 
+reusing them to optimize performance. If you use a dependency injection 
+container for your application, the client instance should be registered with a 
+singleton lifetime.
+
+[discrete]
+[[cloud-deployment]]
+=== Connecting to a cloud deployment
+
+Connecting to an Elasticsearch Service deployment is achieved by providing the 
+unique Cloud ID for your deployment when configuring the `ElasticsearchClient` 
+instance. You can retrieve the Cloud ID from the homepage of the deployment in 
+Elasticsearch Service. You also require suitable credentials that your 
+application uses to authenticate with your deployment.
+
+As a security best practice, it is recommended to create a dedicated API key per 
+application, with permissions limited to only those required for any API calls 
+the application is authorized to make.
+
+The following snippet shows you how to create a client instance that connects to 
+an {es} deployment in the cloud.
+
+[source,csharp]
+----
+using Elastic.Clients.Elasticsearch;
+using Elastic.Transport;
+
+var client = new ElasticsearchClient("<CLOUD_ID>", new ApiKey("<API_KEY>"));
+----
+
+Replace the placeholder string values above with your cloud ID and the API key 
```

Comment:
I think we should make this one of those numbered code comments (not sure what we call them), with the blue numbered circle as this directly references the line of code it relates to?